src/main/java/org/jenkinsci/test/acceptance/docker/DockerContainer.java

```diff
@@ -90,9 +90,10 @@ public String ipBound(int n) {
                 return getIpAddress();
             }
             String out = Docker.cmd("port").add(cid, n).popen().verifyOrDieWith("docker port command failed").trim();
-            if (out.isEmpty())  // expected to return single line like "0.0.0.0:55326"
+            if (out.isEmpty())  // expected to return single line like "0.0.0.0:55326" or [::]:55326
                 throw new IllegalStateException(format("Port %d is not mapped for container %s", n, cid));
-            return out.split(":")[0];
+            // assumes it is published only in ipv6 or only in ipv4
+            return ipv6Enabled() ? out.substring(1,out.lastIndexOf(":") - 1) : out.split(":")[0];
```

Comment:
Why would this need to be conditional? AFAICT both IPv4 and IPv6 formats would work fine by taking the last `:` to be the port suffix, and anything before that to be an IP address.

Reply:
Also need to remove brackets if present as `docker port` output rightfully encloses ipv6 ips in brackets but you are right the conditional can be removed, changed in be883c6824d80b5bcbf8e4244369c9d5bbb708cb